devenv/main.py

```diff
@@ -75,17 +77,62 @@ def devenv(argv: Sequence[str], config_path: str) -> ExitCode:
 
 
 def main() -> ExitCode:
-    import sys
+    # this is also used to see if we're the child process
+    script_logfile = os.environ.get("SCRIPT")
+
+    if script_logfile:
+        return devenv(sys.argv, f"{home}/.config/sentry-devenv/config.ini")
+
+    import tempfile
+
+    _, fp = tempfile.mkstemp()
+    # script (macos/linux) runs the subcommand with a tty, and tees the output to a file.
+    # this way we can very easily capture all output from devenv and send it
+    # to sentry as an attachment if an error occurs.
+    cmd = ("/usr/bin/script", "-qe", fp, *sys.argv)
+
+    # the reason we're subprocessing instead of os.execv(cmd[0], cmd)
+    # is that script must exit (so that the complete log file is committed to disk)
+    # before sentry sends the event...
+    rc = subprocess.call(cmd)
+
+    if rc == 0:
+        return rc
+
+    # i'd love to be able to send a full event in the child then
```

Comment:
I wonder if we can do something something trace id and have both exist? and just link them after the fact?

Reply:
ahhh that might work, i'm thinking we start a transaction in the parent and start a span for the child process then only transaction.finish() if an error happens in the child

not sure what the transaction would look like in sentry ui (like, can we easily see the child error event) but gonna give it a go